RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -19,21 +20,26 @@ public PassParameterByReferenceQuickFix(ParserRuleContext context, QualifiedSele
 
         public override void Fix()
         {
-            string pattern = "^\\s*" + Tokens.ByVal + "(\\s+)";
-            string rgxReplacement = Tokens.ByRef + "$1";
-            Regex rgx = new Regex(pattern);
+            var byValParameter = Context.GetText();
 
-            var parameter = Context.GetText();
-            var newParameter = rgx.Replace(parameter, rgxReplacement);
+            var byRefParameter = BuildByRefParameter(byValParameter);
 
-            var selection = Selection.Selection;
+            ReplaceByValParameterInModule(byValParameter, byRefParameter);
+        }
 
+        private string BuildByRefParameter(string originalParameter)
+        {
+            var everythingAfterTheByValToken = originalParameter.Substring(Tokens.ByVal.Length);
+            return Tokens.ByRef + everythingAfterTheByValToken;
+        }
+        private void ReplaceByValParameterInModule( string byValParameter, string byRefParameter)
+        {
+            var selection = Selection.Selection;
             var module = Selection.QualifiedName.Component.CodeModule;
-            {
-                var lines = module.GetLines(selection.StartLine, selection.LineCount);
-                var result = lines.Replace(parameter, newParameter);
-                module.ReplaceLine(selection.StartLine, result);
-            }
+
+            var lines = module.GetLines(selection.StartLine, selection.LineCount);
+            var result = lines.Replace(byValParameter, byRefParameter);
```

Comment:
FWIW, the corner-cases are the ones that will kill you - @ThunderFrame has been beating that into our collective heads since before I started contributing. Don't get the impression that we're criticising - there's just a ton of twisted but legal syntax to account for. Just imagine the absolute worst possible coding habits, and then account for that.  You'll get the hang of it.  :+1: